crates/typst-pdf/src/color.rs

```diff
@@ -377,26 +378,29 @@ impl PaintEncode for Color {
 /// Extra color space functions.
 pub(super) trait ColorSpaceExt {
     /// Returns the range of the color space.
-    fn range(self) -> [f32; 6];
+    fn range(self) -> &'static [f32];
 
     /// Converts a color to the color space.
-    fn convert<U: QuantizedColor>(self, color: Color) -> [U; 3];
+    fn convert<U: QuantizedColor>(self, color: Color) -> SmallVec<[U; 4]>;
 }
 
 impl ColorSpaceExt for ColorSpace {
-    fn range(self) -> [f32; 6] {
-        [0.0, 1.0, 0.0, 1.0, 0.0, 1.0]
+    fn range(self) -> &'static [f32] {
+        match self {
+            ColorSpace::Cmyk => &[0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0],
+            ColorSpace::D65Gray => &[0.0, 1.0],
+            _ => &[0.0, 1.0, 0.0, 1.0, 0.0, 1.0],
```

Comment:
Maybe it's actually better to do an exhaustive match here to prevent strife in the future

Reply:
Good point! Thanks for that! I've changed [these lines](https://github.com/typst/typst/pull/4568/files#diff-55650bf2dd7b33ec02c37fd3b314eb3a5db0c2aa6bf0baf269ef9c03fe716c95R389-R397) to do an exhaustive search.